frontend/src/interfaces/RoomInterface.tsx

```diff
@@ -0,0 +1,28 @@
+export interface CreateRoomState {
+    title: string;
+    password: string;
+    shouldJoin: boolean;
+    isPrivate: boolean;
+    maxAmount: number | '';
+    dateOfExchange: string;
+    rules: string;
+}
+
+export interface JoinRoomState {
+    roomUrl: string;
+    password: string;
+}
+
+export interface RoomState {
```

Comment:
Would probably make more sense to name this more appropriately, `RoomDB` if this is the data received from the DB.